src/Filecoin/Transaction.h

```diff
@@ -26,7 +26,7 @@ class Transaction {
     uint256_t value;
     // Miner fee
     uint256_t gasPrice;
-    uint256_t gasLimit;
+    int64_t gasLimit;
```

Comment:
can we keep it as uint64? sicne we do casting in  `Cbor::Encode::uint((uint64_t)gasLimit)` in `Filecoin/Transaction.cpp` ?

Reply:
@vikmeup The Filecoin Lotus implementation also defines this parameter as `int64`. I'm not sure why they do that, maybe negative values have some special meaning in the future (like unlimited). It is cast to `uint64_t` again because CBOR specifies signedness as part of the data type, not via two's-complement in the number.